### PR TITLE
feat: Query packages and milestones, variables on all fields, and QueryTypeMux

### LINKED ISSUE #7: Support querying packages

### LINKED ISSUE #13: Support querying milestones

### LINKED ISSUE #38: Support variables in all query fields

### LINKED ISSUE #39: Use QueryTypeMux instead of map with QueryTypeFunc
https://pkg.go.dev/github.com/grafana/grafana-plugin-sdk-go@v0.77.0/backend/datasource?tab=doc#QueryTypeMux

### PR DESCRIPTION
**Pull request by kminehart**
* Moved from a map and self-routing to QueryTypeMux ( fixes #39 )
* Added ability to use variables on all fields ( fixes #38 )
* Added ability to query packages (docker, npm, maven, pypi) ( fixes #7 )
* Added ability to query Milestones ( fixes #13 )

### CODE REVIEW

**Comment by ryantxu** [comment]
you always make these so easy to review :)  
image

anything in particular you want feedback on?

File: go.sum [hunk 1]
```diff
@@ -125,8 +125,12 @@ github.com/prometheus/procfs v0.0.8 h1:+fpWZdT24pJBiqJdAwYBjPSk+5YmQzYNPYzQsdzLk
 github.com/prometheus/procfs v0.0.8/go.mod h1:7Qr8sr6344vo1JqZ6HhLceV9o3AJ1Ff+GxbHq6oeK9A=
 github.com/shurcooL/githubv4 v0.0.0-20200414012201-bbc966b061dd h1:EwtC+kDj8s9OKiaStPZtTv3neldOyr98AXIxvmn3Gss=
 github.com/shurcooL/githubv4 v0.0.0-20200414012201-bbc966b061dd/go.mod h1:hAF0iLZy4td2EX+/8Tw+4nodhlMrwN3HupfaXj3zkGo=
+github.com/shurcooL/githubv4 v0.0.0-20200802174311-f27d2ca7f6d5 h1:CA6Mjshr+g5YHENwllpQNR0UaYO7VGKo6TzJLM64WJQ=
+github.com/shurcooL/githubv4 v0.0.0-20200802174311-f27d2ca7f6d5/go.mod h1:hAF0iLZy4td2EX+/8Tw+4nodhlMrwN3HupfaXj3zkGo=
```
Comment (ryantxu) [inline go.sum:129]: i know it does not really make a difference, but can you update the go.sum so only the current files are listed?  better than a long list of every version that was ever included!

File: src/types.ts [hunk 2]
```diff
@@ -45,38 +57,45 @@ export enum IssueTimeField {
   ClosedAt,
 }
 
-export interface ReleasesOptions {}
-export interface TagsOptions {}
-export interface PullRequestsOptions {
+export interface Indexable {
```
Comment (ryantxu) [inline src/types.ts:60]: um... this kinda sidesteps all value in using typescript!

**Comment by kminehart** [comment]
yeah I just kind of went in the zone. I think you hit the nail on the head, I'm least confident about how I handled "supporting variables on all keys" (src/variables.ts)

File: go.sum (hunk 1 above)
Comment (kminehart) [reply go.sum:133]: will go mod tidy suffice?

File: src/types.ts (hunk 2 above)
Comment (kminehart) [reply src/types.ts:60]: yeah, that's the only way I could accomplish this (as far as I know :\ )
https://github.com/grafana/github-datasource/pull/42/files#diff-f65c1b6588a53320c1abf59e0caed96cR9-R26

any suggestions?

**Review by scottlepp — COMMENTED** [review]
A bit confused with the mux handlers pattern.  We had used this pattern in the past and went away from it with the direction to use the datasource query method handing the queries.  It would be nice to be consistent.  @ryantxu was this not the direction?

If we do use this pattern, does the handler need to delegate to the datasource?  I prefer single responsibility, so it seems like the code from the datasource could be moved to the handlers.

**Comment by scottlepp** [comment]
Also, can we try and break these into smaller prs?  Hard to review.

**Comment by kminehart** [comment]
I'm going to merge this for now and create an issue for refactoring the template variables. I tried replacing the 'Indexable' interface with something that allowed each `Option` interface define how they use TemplateSrv to replace variables and it fell flat. I'll have to keep experimenting with it.

**Pull request merged**